app/src/services/cpk.ts

```diff
@@ -417,6 +433,65 @@ class CPKService {
       throw err
     }
   }
+
+  redeemPositions = async ({
+    isConditionResolved,
+    questionId,
+    numOutcomes,
+    winningOutcome,
+    oracle,
+    collateralToken,
+    marketMaker,
+    conditionalTokens,
+  }: CPKRedeemParams): Promise<TransactionReceipt> => {
+    try {
+      const signer = this.provider.getSigner()
+      const account = await signer.getAddress()
+
+      const transactions = []
+      if (!isConditionResolved) {
+        transactions.push({
+          operation: CPK.CALL,
+          to: oracle.address,
+          value: 0,
+          data: OracleService.encodeResolveCondition(questionId, numOutcomes),
+        })
+      }
+
+      const conditionId = await marketMaker.getConditionId()
+
+      transactions.push({
```

Comment:
Wouldn't this need to be inside `if (winningOutcome)`? I think it doesn't make sense to spend gas redeeming losing tokens.

Reply:
Oh, nevermind, disregard that and just merge this.